tensorflow_datasets/core/dataset_utils.py

```diff
@@ -268,11 +277,12 @@ def as_numpy(dataset, graph=None):
   return tf.nest.pack_sequence_as(nested_ds, flat_np)
 
 
-def dataset_shape_is_fully_defined(ds):
+def dataset_shape_is_fully_defined(ds: tf.data.Dataset) -> bool:
   output_shapes = tf.compat.v1.data.get_output_shapes(ds)
   return all([ts.is_fully_defined() for ts in tf.nest.flatten(output_shapes)])
 
 
-def features_shape_is_fully_defined(features):
+def features_shape_is_fully_defined(
+    features: features_lib.FeaturesDict) -> bool:
```

Comment:
I think it should be a `features_lib.FeatureConnector`

Reply:
Yes got it. Changed.